### PR TITLE
Add Embeded wind/solar and pumped hydro storage to ELEXON parser.

### PR DESCRIPTION
**Pull request by unitrium**
## Issue
The data provided by the BMRS FUEL_INST report does not describe the full power generation in the UK. To complete it we need to add embedded wind/solar accounting for around 26% of installed capacity and hydro storage.

Related issue #4236.
## Description

Use the National Grid Eso database to query and add the missing data.

### CODE REVIEW

**Comment by pierresegonne** [comment]
Do you have a reference for the issue that you're mentioning?

File: parsers/ELEXON.py [hunk 1]
```diff
@@ -224,11 +271,30 @@ def parse_production_FUELINST(
...
+    df.rename(columns=ESO_FUEL_MAPPING, inplace=True)
+    return df.set_index("datetime")
+
 
+def process_production_events(
+    fuel_inst_data: pd.DataFrame, eso_data: pd.DataFrame
+) -> List[dict]:
+    """Combine FUELINST report and ESO data together to get the full picture and to EM Format."""
+    df = fuel_inst_data.join(eso_data, rsuffix="_eso")
+    df.rename(columns={"wind_eso": "wind", "solar_eso": "solar"}, inplace=True)
```
Comment (pierresegonne) [inline parsers/ELEXON.py:293]: nitpick, but we generally avoid doing inplace operations

**Comment by VIKTORVAV99** [comment]
It should be the last part to fix #4835, at least temporarily.

File: parsers/ELEXON.py (hunk 1 above)
Comment (pierresegonne) [reply parsers/ELEXON.py:293]: So that means we discard completely wind and solar from bnreports in favour of this new source?

**Review by pierresegonne — APPROVED** [review]
LGTM but would be nice to get more context when describing the PR

File: parsers/ELEXON.py (hunk 1 above)
Comment (pierresegonne) [reply parsers/ELEXON.py:293]: Elements for why inplace is not really recommended in most cases:

https://towardsdatascience.com/why-you-should-probably-never-use-pandas-inplace-true-9f9f211849e4


TL;DR is that it creates mutability issues

Comment (unitrium) [reply parsers/ELEXON.py:293]: @pierresegonne we need to investigate which source we keep, once BNReports comes back online. The source might actually not report embedded wind and solar as well.
